src/Stratis.Bitcoin.Api.Tests/ApiSettingsTest.cs

```diff
@@ -308,12 +258,21 @@ public void GivenUseHttpsAndNoCertificateFilePath_ThenShouldThrowConfigurationEx
             // Act.
             var settingsAction = new Action(() =>
                 {
-                    new FullNodeBuilder().UseNodeSettings(nodeSettings).UseApi().Build()
-                            .NodeService<ApiSettings>();
+                    FullNodeSetup(nodeSettings);
                 });
 
             // Assert.
             settingsAction.Should().Throw<ConfigurationException>();
         }
+
+        private static ApiSettings FullNodeSetup(NodeSettings nodeSettings)
```

Comment:
What is this doing in this PR?

Reply:
> Commit a007c27 is Part 2.
Part 3 commit, I will focus on fixing broken integration tests.

This was related to Part 2 of the commit, mentioned in the above comments.

I needed to get the unit tests working against the PR otherwise the first part would have been merged into master with failing tests.

I have tried to keep the work as separate as possible,

Reply:
I can see a lot of changes which seem to have nothing to do with this PR. Create separate PRs, not just separate commits. Obviously, you must change the tests to prevent failing tests, but that doesn't mean you need to refactor unrelated code. If you want to refactor something, create separated PR.

Reply:
Yes, and you could find similar problems in 100 different files all over the code base and they have nothing to do with this PR and just waste my time if I am about to review this PR. At some point I will start rejecting pull requests which contain even single line that is not related to the issue

Reply:
Yes, problem is finding a way to create separate PR's when unit tests are impacted, like in this case.

I'll check with the guys the best approach on keeping the PR's down to bit size pieces using the appropriate git commands.

Reply:
So the idea is that when you make a change to the code, it will require some changes in tests. Those changes are part of the PR. However beautifying the code even in the same class is not. So just change the test with minimal change in order to have it working. Then if you feel another change should be made to make something more beautiful, make separate PR